activesupport/test/subscriber_test.rb

```diff
@@ -75,4 +99,16 @@ def test_detaches_subscribers
     assert_equal [], TestSubscriber2.events
     assert_equal 1, TestSubscriber.events.size
   end
+
+  def test_detaches_subscribers_from_specific_events
+    ActiveSupport::Notifications.instrument("open_party.doodle")
+    ActiveSupport::Notifications.instrument("another_open_party.doodle")
+
+    assert_equal 1, TestSubscriber3.events.size
+    assert_equal "another_open_party.doodle", TestSubscriber3.events.first.name
+  end
+
+  def test_detach_from_does_not_remove_subscriber_if_only_detaching_from_some_events
+    assert_equal 2, ActiveSupport::Subscriber.subscribers.count
```

Comment:
I think this test is prone to flakyness, if there is another test (in another file) that adds a subscriber, the count won't be 2 anymore. Would be better to check if `detach_from` can be called multiple times in a row and it correctly unsubscribe the events

Reply:
So I've left the intent of this test the same (ie. ensure that if we have a subscriber detaching from some but not all events, the subscriber is still tracked by `ActiveSupport::Subscriber`, but I've modified the test to use `find` instead of asserting on count, which is flaky like you mentioned!

I also added another test to test what you suggested
> check if detach_from can be called multiple times in a row and it correctly unsubscribe the events

which tests that 
a) we can use `detach_from:` multiple times in a row to correctly unsubscribe from events, and
b) if a subscriber detaches from all events, it is no longer tracked by `ActiveSupport::Subscriber`.

Let me know what you think!